### PR TITLE
UnitsAggregationSorting inheritance

### LINKED ISSUE #947: UnitsAggregationSorting fails to inherit property
Hi,

I saw in the `UnitsAggregationSorting` code that there is a copying of properties if they are consistent over all sortings.

However, look at the following code:

```python
sorting.annotate(name="Bonjour")
sorting.get_annotation("name")
> 'Bonjour'

sorting1 = sorting.select_units([0, 1])
sorting1.get_annotation("name")
> 'Bonjour'

sorting2 = sorting.select_units([unit_id for unit_id in sorting.unit_ids if unit_id != 0 and unit_id != 1])
sorting2.get_annotation("name")
> 'Bonjour'

new_sorting = si.UnitsAggregationSorting([sorting1, sorting2])
new_sorting.get_annotation("name")
> None
```

I believe the property should be inherited in this case.

Thanks,
Aurélien W.

### PR DESCRIPTION
**Pull request by Dradeliomecus**
Fixes #947

### CODE REVIEW

**Comment by Dradeliomecus** [comment]
To know if the `missing_value` for setting properties can be inferred from the dtype kind, I was thinking of making the varaible `default_missing_values` (in `BaseExtractor.set_property`) a class property that can be accessed by the `UnitsAggregationSorting`.

What do you think @samuelgarcia ?

**Comment by samuelgarcia** [comment]
OK for me.

File: spikeinterface/core/unitsaggregationsorting.py [hunk 1]
```diff
@@ -60,25 +61,36 @@ def __init__(self, sorting_list, renamed_unit_ids=None):
...
-        property_keys = sorting_list[0].get_property_keys()
+        property_keys = {}
         property_dict = {}
+        deleted_keys = []
+        for sort in sorting_list:
+            for prop_name in sort.get_property_keys():
+                if prop_name in deleted_keys:
+                    continue
+                if prop_name in property_keys:
+                    if property_keys[prop_name] != sort.get_property(prop_name).dtype:
```
Comment (alejoe91) [inline spikeinterface/core/unitsaggregationsorting.py:72]: wouldn't this fail??? Now `property_keys` is initialized as an empty dict!

Comment (alejoe91) [reply spikeinterface/core/unitsaggregationsorting.py:72]: @Dradeliomecus can you add a test for this?

Comment (Dradeliomecus) [reply spikeinterface/core/unitsaggregationsorting.py:72]: Done.

**Comment by alejoe91** [comment]
Closes https://github.com/SpikeInterface/spikeinterface/issues/947

**Pull request merged**